### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by victor-ab** [thongn98/receipt-information-extraction#1]
Hi!

First of all, thank you very much for pulling this together, I was looking for this task on the Microsoft repo and you saved me a lot of time.

My question is: What about the image embedding, where is it placed in the code? I thought we'd need the jpg itself to embed it and use it as an input for the model. Where did I get it wrong?
I only see you using the image to get its size for the position embedding.

**Comment by thongn98** [comment]
Hi Victor,

Thank you for checking out the repo and for your great question. 

From my understanding of the LayoutLM paper and its repo, you can use image embedding (with some modification to their model code; they also mentioned that they used Resnet-101 as backbone their Faster R-CNN [here](https://github.com/microsoft/unilm/issues/165) ) with LayoutLM embedding to do the task. However, with some time limitation and from reading the other issues on their repository (and I guess from seeing the actual decent result when I did my initial training), I decided to only use the LayoutLM embedding we get from the model to do the token classification. I'm pretty sure the example on LayoutLM's repo did the same thing with the token classification task on the FUNSD dataset. Adding the image embedding is something I'd love to try to see if the model can achieve better result with it, which I won't be surprised if it does. Feel free leave another comment here or email me if you decide to try it out! I'd love to see the result

**Comment by victor-ab** [comment]
Got it!
Thank you so much for the response.

Yes, just now I saw them suggesting to integrate a Faster R-CNN implementation with the LayoutLM output. It seems like a good Idea. I'll try to implement it later, and I'll share if I have any luck.

Furthermore, I was looking into the dataset you [shared](https://drive.google.com/drive/folders/1ShItNWXyiY1tFDM5W02bceHuJjyeeJl2), but I only found the test images `task3-test (347p)` for the task 3, not training data nor their json data with the bounding boxes and the texts. Do you know where is it?

**Comment by thongn98** [comment]
Hi Victor, 

Sorry for the late response. So it's a bit confusing how the dataset is set up. The training label for task 3 is actually in `0325updated.task2train(626p)` but you'll have to label each word (token) separately if you want to approach it as a token classification problem. I have outlined the process I used to label the data in the readme. I did have to go in there partially manually to label the data and the repository does contain the labeled training data in a `.txt` file in the `data/` directory (only the first ~20000 tokens/words out of ~70000 from 626 images). As for the test label, you'll have to submit the model to the competition itself to get the accuracy as the competition doesn't publicly provide the test label. 

Hope that helps !
